pkg/rpc/context.go

```diff
@@ -1079,7 +1079,7 @@ type delayingConn struct {
 	readBuf     *bytes.Buffer
 }
 
-func (d delayingConn) Write(b []byte) (n int, err error) {
+func (d *delayingConn) Write(b []byte) (n int, err error) {
 	tNow := timeutil.Now()
 	if d.lastSendEnd.Before(tNow) {
 		d.lastSendEnd = tNow
```

Comment:
This would have done nothing before your change. Is that a problem? I don't remember what we use `lastSendEnd` for, but this really wouldn't have worked whatsoever unless we only use it down below in this same function to set `ReadTime`.

Reply:
I think it's more than `lastSendEnd` that's an issue though, as here:

https://github.com/cockroachdb/cockroach/blob/397046b4aca08f01bfefaf1856041094ea4f35ff/pkg/rpc/context.go#L1093-L1095

we're writing the header to the `d.Conn`.  Without the pointer receiver, won't that update not get persisted?

Reply:
No, that'll work fine, because `d.Conn` is an interface and therefore a pointer. So even a copy of `d` will have the same `Conn` as the original `d`. The only issue here is value (non-pointer) fields in `d`, such as `d.lastSendEnd`.

Reply:
Ahh, cool, thanks.  I didn't realize that interfaces were automatically pointers, but that makes sense.

OK, I think we're good leaving this as-is then, as `lastSendEnd` is only used in this function, and then written to the `hdr` which then gets written to the `Conn` interface.  I've reverted the change.  I think we're good to go now.